pkg/controller/baremetalhost/host_state_machine.go

```diff
@@ -206,6 +206,11 @@ func (hsm *hostStateMachine) handleRegistrationError(info *reconcileInfo) action
 		hsm.NextState = metal3v1alpha1.StateRegistering
 		return actionComplete{}
 	}
+	if hsm.Host.Spec.Image != nil {
```

Comment:
If credentials are wrong, and image exists, this code will constantly retry registration, isn't it?

Reply:
Yes, that's correct. We have another PR up to introduce similar retry behavior for other failures.